### PR TITLE
Bug with some lambdas and json_go

### PR DESCRIPTION
**Pull request by ldemailly**
- [x] test
- [x] fix
- [x] review comments from earlier PRs

### CODE REVIEW

File: main_test.txtar [hunk 1]
```diff
@@ -267,6 +267,12 @@ stderr '^func level3\(\){level2\(\)}$'
 stderr 'Total 1 error'
 stderr '^<err: no PLUS on left=1 right="x" in func level1\(\){1\+"x"}>$'
 
+# json with some functions
+grol -quiet -c 'f=()=>{{"k":"v"}}; println(json_go(f))'
+!stderr 'json: unsupported type'
+!stdout 'json: unsupported type'
+stdout '^"\(\)=>{{\\"k\\":\\"v\\"}}"$'
```
Comment (ldemailly) [inline main_test.txtar:274]: regexp escaping is getting a bit tough/annoying, the output is
```
"()=>{{\"k\":\"v\"}}"
```

File: extensions/extension.go [hunk 2]
```diff
@@ -458,21 +463,21 @@ func loadFunc(env any, _ string, args []object.Object) object.Object {
...
 	all, err := io.ReadAll(f)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
 	}
 	// Eval the content.
 	res, err := eval.EvalString(env, string(all), false)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
```
Comment (ldemailly) [inline extensions/extension.go:480]: @ccoVeille I bet you probably don't like that name, suggestion for a better one?

Comment (ccoVeille) [reply extensions/extension.go:480]: I do dislike it.

ReportError ?

ReturnError?

Comment (ldemailly) [reply extensions/extension.go]: overlord suggested a few, I picked ConvertError() but I could be convinced to use something else

not ReportError as it's not what it does... it makes an error object (which cause evaluation to abort)
```go
return s.ReturnError()
```
looks a bit stuttering :)

Comment (ldemailly) [reply extensions/extension.go]: maybe I can make the other one ErrorString() and use Error for this one

Comment (ldemailly) [reply extensions/extension.go]: I think this is actually decent now:
```go
/ NewError creates a new error object from a plain string.
// NewError will attach the stack trace to the NewError object.
func (s *State) NewError(msg string) object.Error
// Errorf formats an object.Error with the given format and args.
func (s *State) Errorf(format string, args ...interface{}) object.Error 
// Error converts from a go error to an object.Error.
func (s *State) Error(err error) object.Error 
```

**Review by ccoVeille — APPROVED** [review]

**Pull request merged**